torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -566,21 +566,21 @@ struct NCCLTraceBuffer {
       duration = getDurationFromFirstEvent(*startEvents, *endEvents);
       guard.lock();
 
-      // Refresh the entry ref, see if it has been overwritten
-      entry = entries_.at(*id % max_entries_);
```

Comment:
@yifuwang the bug is in this line. the old code would invoke operator= and copy the requested entry contents into the value of the old reference, which could now belong to a different entry.  The new code would assign a new pointer value locally only, and let us check if it's still pointing to the same entry or not.  (it's a ring buffer)

Reply:
Ah I see it now! Thanks for highlighting